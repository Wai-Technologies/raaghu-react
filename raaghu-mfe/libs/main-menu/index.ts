export { default as MainMenu } from "./main-menu";
export {default as PageNotFound} from './pagenotfound';