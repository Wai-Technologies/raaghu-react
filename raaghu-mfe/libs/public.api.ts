export * from "./state-management/public.api";
export * from "./shared/public.api";