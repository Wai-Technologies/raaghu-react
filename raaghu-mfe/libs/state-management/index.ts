import tagAdminReducer from "./tag-admin/tag-admin-slice";
import fileDescriptorsReducer from "./file-descriptors/file-descriptors-slice";
import directoryDescriptorsReducer from "./directory-descriptors/directory-descriptors-slice";
import auditLogsReducer from "./audit-logs/audit-logs-slice";
import languageTextsReducer from "./language-texts/language-texts-slice";
import pollAdminReducer from "./poll-admin/poll-admin-slice";
import planAdminReducer from "./plan-admin/plan-admin-slice";
import roleReducer from "./role/role-slice";
import organizationUnitReducer from "./organization-unit/organization-unit-slice";
import permissionsReducer from "./permissions/permissions-slice";
import scopesReducer from "./scopes/scopes-slice";
import featuresReducer from "./features/features-slice";
import { useDispatch } from "react-redux";
import { persistReducer, persistStore } from "redux-persist";
import { configureStore, combineReducers } from "@reduxjs/toolkit";
import storage from "redux-persist/lib/storage";
import languagesReducer from "./languages/languages-slice";
import PageNotFoundReducer from "./pagenotfound/pagenotfound-slice";
import formsResponseReducer from "./formsresponse/formsresponse-slice";
import HomeReducer from "./home/home-slice";
import logoutReducer from "./logout/logout-slice";
import registerReducer from "./register/register-slice";
import blogPostReducer from "./blog-post/blog-post-slice";
import pagesReducer from "./pages/pages-slice";
import { forgotPasswordReducer } from "./forgot-password/forgotpassword-slice";
import editionReducer from "./edition/edition-slice";
import { subscriptionReducer } from "./subscription/subscription-slice";
import languageTextReducer from "./language-text/language-text-slice";
import organizationReducer from "./organization-tree/organization-tree-slice";
import rolesReducer from "./role/role-slice";
import ClaimTypeReducer from "./claim-type/claim-type-slice";
import securityLogsReducer from "./security-logs/security-logs-slice";
import userReducer from "./user/user-slice";
import settingsReducer from "./settings/settings-slice";
import FileManagementReducer from "./file-management/file-management-slice";
import hostReducer from "./host/host-slice";
import applicationsReducer from "./applications/applications-slice";
import urlForwardingReducer from "./url-forwarding/url-forwarding-slice";
import textTemplateReducer from "./text-template/text-template-slice";
import tenantReducer from "./tenant/tenant-slice";
import chatsReducer from "./chats/chats-slice";
import bloggerReducer from "./blogger/blogger-slice";
import formsReducer from "./forms/forms-slice";
import paymentPlansReducer from "./payment-plans/paymentPlans-slice";
import personalDataReducer from "./personal-data/personal-data-slice";
import commentsReducer from "./comments/comments-slice";
import tagsReducer from "./tags/tags-slice";
import myAccountReducer from "./my-account/my-account-slice";
import paymentRequestsReducer from "./payment-requests/paymentRequests-slice";
import newslettersReducer from "./newsletters/newsletters-slice";
import pollsReducer from "./polls/polls-slice";
import globalResourcesReducer from "./global-resources/globalResources-slice";
import menusReducer from "./menus/menus-slice";
import blogsReducer from "./Blogs/blogs-slice";
import configureReducer from "./config/config-slice";

const persistConfig = {
    key: "root",
    storage,
    blacklist: ["forgotPassword"],
};
const rootReducer = combineReducers({
    tagAdmin: tagAdminReducer,
    fileDescriptors: fileDescriptorsReducer,
    directoryDescriptors: directoryDescriptorsReducer,
    auditLogs: auditLogsReducer,
    languageTexts: languageTextsReducer,
    pollAdmin: pollAdminReducer,
    planAdmin: planAdminReducer,
    role: roleReducer,
    organizationUnit: organizationUnitReducer,
    permissions: permissionsReducer,
    features: featuresReducer,
    languages: languagesReducer,
    configureStore: configureReducer,
    PageNotFound: PageNotFoundReducer,
    formsResponse: formsResponseReducer,
    Home: HomeReducer,
    Logout: logoutReducer,
    register: registerReducer,
    blogPost: blogPostReducer,
    pages: pagesReducer,
    forgotPassword: forgotPasswordReducer,
    subscription: subscriptionReducer,
    edition: editionReducer,
    languagesText: languageTextReducer,
    organization: organizationReducer,
    roles: rolesReducer,
    claimTypes: ClaimTypeReducer,
    securityLogs: securityLogsReducer,
    applications: applicationsReducer,
    scopes: scopesReducer,
    textTemplate: textTemplateReducer,
    auditLog: auditLogsReducer,
    user: userReducer,
    settings: settingsReducer,
    tenant: tenantReducer,
    chats: chatsReducer,
    blogger: bloggerReducer,
    forms: formsReducer,
    fileManagement: FileManagementReducer,
    paymentPlans: paymentPlansReducer,
    urlForwarding: urlForwardingReducer,
    personalData: personalDataReducer,
    comments: commentsReducer,
    tags: tagsReducer,
    host: hostReducer,
    myaccount: myAccountReducer,
    paymentRequests: paymentRequestsReducer,
    polls: pollsReducer,
    globalResources: globalResourcesReducer,
    newsletters: newslettersReducer,
    menus: menusReducer,
    blogs: blogsReducer
});
const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: { persistedReducer },
});
export const persistor = persistStore(store);
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;
