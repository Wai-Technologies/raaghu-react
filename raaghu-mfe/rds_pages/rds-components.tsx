export * from "../../raaghu-components/src/index";
