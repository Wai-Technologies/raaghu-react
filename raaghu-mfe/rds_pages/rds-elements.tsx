export * from "../../raaghu-elements/src/index";
