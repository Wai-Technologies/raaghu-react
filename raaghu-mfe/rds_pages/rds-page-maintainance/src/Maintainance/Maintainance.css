.mobile-btn{
  display: none;
}
.desktop-btn{
  display: flex;
}
.custom-z-index{
  z-index: 1;
}

@media (max-width: 767px){
  .mobile-btn{
      display: block;
  }
  .desktop-btn{
      display: none;
  }
  .card.h-100 {
      height: calc(100vh - 90px) !important;
  }
}