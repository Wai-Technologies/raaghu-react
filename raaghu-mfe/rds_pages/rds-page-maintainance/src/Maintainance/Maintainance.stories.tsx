import React from "react";
import { ComponentStory, ComponentMeta, addDecorator } from "@storybook/react";
import RdsPageMaintainance from "./Maintainance";

export default {
    title: "Pages/Maintainance",
    component: RdsPageMaintainance,
} as ComponentMeta<typeof RdsPageMaintainance>;

const Template: ComponentStory<typeof RdsPageMaintainance> = () => (
    <RdsPageMaintainance />
);

export const Default = Template.bind({});

Default.args = {};
