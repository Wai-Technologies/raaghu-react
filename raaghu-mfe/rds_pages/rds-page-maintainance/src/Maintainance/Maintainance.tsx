import React, { useEffect, useRef, useState } from "react";
import {
    RdsCompWebsiteLog,
    RdsCompCache,
    RdsCompAlertPopup,
} from "../../../rds-components";
import "./Maintainance.css";
import {
    RdsButton,
    RdsFabMenu,
    RdsIcon,
    RdsNavtabs,
} from "../../../rds-elements";
import { useTranslation } from "react-i18next";

const cachedata = [
    { name: "AbpUserSettingsCache", id: 1 },
    { name: "AbpZeroRolePermissions", id: 2 },
    { name: "AbpZeroTenantCache", id: 3 },
    { name: "AbpZeroEditionFeatures", id: 4 },
];
const Maintainance = () => {
    const [tabcache, setTabcache] = useState(true);
    const [WebsiteLog, setWebsiteLog] = useState(false);
    const [activeNavTabId, setActiveNavTabId] = useState("nav-cache");
    const [cache, setcache] = useState(cachedata);
    const { t } = useTranslation();

    const listItems1 = [
        {
            value: "Clear All",
            some: "value",
            key: "clearall",
            icon: "delete",
            iconWidth: "20px",
            iconHeight: "20px",
        },
    ];

    const listItems2 = [
        {
            value: "Refresh",
            some: "value",
            key: "refresh",
            icon: "refresh",
            iconWidth: "20px",
            iconHeight: "20px",
        },
        {
            value: "Download All",
            some: "value",
            key: "downloadall",
            icon: "download_data",
            iconWidth: "20px",
            iconHeight: "20px",
        },
    ];

    const navtabsItems = [
        {
            label: t("Cache"),
            id: "nav-cache",
            tablink: "#nav-cache",
            ariacontrols: "nav-cache",
        },
        {
            label: t("WebsiteLog"),
            id: "nav-websiteLogs",
            tablink: "#nav-websiteLogs",
            ariacontrols: "nav-websiteLogs",
        },
    ];

    const websiteLogData = [
        {
            status: "INFO",
            content:
                "2021-12-28 15:43:39,431 [1 ] oft.EntityFrameworkCore.Model.Validation - Entity &#39;Edition&#39; has a global query filter defined and is the required end of a relationship with the entity &#39;EditionFeatureSetting&#39;. This may lead to unexpected results when the required entity is filtered out. Either configure the navigation as optional, or define matching query filters for both entities in the navigation. See https://go.microsoft.com/fwlink/?linkid=2131316 for more information.",
        },
        {
            status: "WARN",
            content:
                "2021-12-28 15:43:39,624 [1 ] oft.EntityFrameworkCore.Model.Validation - No type was specified for the decimal property &#39;DailyPrice&#39; on entity type &#39;SubscribableEdition&#39;. This will cause values to be silently truncated if they do not fit in the default precision and scale. Explicitly specify the SQL server column type that can accommodate all the values in &#39;OnModelCreating&#39; using &#39;HasColumnType()&#39;, specify precision and scale using &#39;HasPrecision()&#39; or configure a value converter using &#39;HasConversion()&#39;.",
        },
        {
            status: "INFO",
            content:
                "2021-12-28 15:43:39,431 [1 ] oft.EntityFrameworkCore.Model.Validation - Entity &#39;Edition&#39; has a global query filter defined and is the required end of a relationship with the entity &#39;EditionFeatureSetting&#39;. This may lead to unexpected results when the required entity is filtered out. Either configure the navigation as optional, or define matching query filters for both entities in the navigation. See https://go.microsoft.com/fwlink/?linkid=2131316 for more information.",
        },
        {
            status: "ERROR",
            content:
                "2021-12-28 15:43:39,624 [1 ] oft.EntityFrameworkCore.Model.Validation - No type was specified for the decimal property &#39;DailyPrice&#39; on entity type &#39;SubscribableEdition&#39;. This will cause values to be silently truncated if they do not fit in the default precision and scale. Explicitly specify the SQL server column type that can accommodate all the values in &#39;OnModelCreating&#39; using &#39;HasColumnType()&#39;, specify precision and scale using &#39;HasPrecision()&#39; or configure a value converter using &#39;HasConversion()&#39;.",
        },
        {
            status: "WARN",
            content:
                "2021-12-28 15:43:39,431 [1 ] oft.EntityFrameworkCore.Model.Validation - Entity &#39;Edition&#39; has a global query filter defined and is the required end of a relationship with the entity &#39;EditionFeatureSetting&#39;. This may lead to unexpected results when the required entity is filtered out. Either configure the navigation as optional, or define matching query filters for both entities in the navigation. See https://go.microsoft.com/fwlink/?linkid=2131316 for more information.",
        },
        {
            status: "INFO",
            content:
                "2021-12-28 15:43:39,624 [1 ] oft.EntityFrameworkCore.Model.Validation - No type was specified for the decimal property &#39;DailyPrice&#39; on entity type &#39;SubscribableEdition&#39;. This will cause values to be silently truncated if they do not fit in the default precision and scale. Explicitly specify the SQL server column type that can accommodate all the values in &#39;OnModelCreating&#39; using &#39;HasColumnType()&#39;, specify precision and scale using &#39;HasPrecision()&#39; or configure a value converter using &#39;HasConversion()&#39;.",
        },
        {
            status: "ERROR",
            content:
                "2022-06-20 20:56:34,313 [4 ] Microsoft.AspNetCore.Hosting.Diagnostics - Request starting HTTP/2 GET https://localhost:44301/AbpUserConfiguration/GetAll?d=1655738793955 application/json -;. This will cause values to be silently truncated if they do not fit in the default precision and scale. Explicitly specify the SQL server column type that can accommodate all the values in &#39;OnModelCreating&#39; using &#39;HasColumnType()&#39;, specify precision and scale using &#39;HasPrecision()&#39; or configure a value converter using &#39;HasConversion()&#39;.",
        },
    ];

    const onchangetabs = (activeNavTabId: any) => {
        setActiveNavTabId(activeNavTabId);
        setWebsiteLog(false);
        if (activeNavTabId == "nav-cache") {
            setTabcache(true);
            setWebsiteLog(false);
        } else {
            setTabcache(false);
            setWebsiteLog(true);
        }
    };
    const Delete = () => { };
    const refreshData = () => { };

    const DeleteAllCacheData = () => {
        setcache([]);
    };

    const downloadcsv = () => {
        const keys = Object.keys(websiteLogData[0]);
        const csvString = websiteLogData
            .map((row: any) => keys.map((key) => row[key]).join(","))
            .join("\n");

        // Create a hidden link element
        const link = document.createElement("a");
        link.style.display = "none";
        link.setAttribute(
            "href",
            "data:text/csv;charset=utf-8," + encodeURIComponent(csvString)
        );
        link.setAttribute("download", "operation_logs.csv");

        // Append the link to the DOM
        document.body.appendChild(link);

        // Click the link to initiate the download
        link.click();

        // Remove the link from the DOM
        document.body.removeChild(link);
    };

    return (
        <div>
            <div className="row">
                <div className="col-md-12">
                    <div className="card px-4 py-4 border-0 rounded-0 card-full-stretch">
                        <div className="row">
                            <div className="col-md-12 text-end pb-3">
                                {tabcache && (
                                    <div className="d-flex justify-content-end">
                                        <div className="desktop-btn">
                                            <RdsButton
                                                type={"button"}
                                                colorVariant="primary"
                                                size="small"
                                                tooltipPlacement="top"
                                                label={t("Clear All") || ""}
                                                icon="delete"
                                                iconColorVariant="light"
                                                iconHeight="12px"
                                                iconWidth="12px"
                                                iconFill={false}
                                                iconStroke={true}
                                                databstarget="#maintainance"
                                                databstoggle="modal"
                                            ></RdsButton>
                                            <RdsCompAlertPopup
                                                alertID="maintainance"
                                                onSuccess={DeleteAllCacheData}
                                            ></RdsCompAlertPopup>
                                        </div>
                                        <div
                                            className="mobile-btn position-fixed bottom-0 end-0 mb-3 me-5 custom-z-index">
                                            <RdsFabMenu
                                                listItems={listItems1}
                                                colorVariant="primary"
                                                menuIcon="plus"
                                                menuiconWidth="12px"
                                                menuiconHeight="12px"
                                            ></RdsFabMenu>
                                        </div>
                                    </div>
                                )}

                                {WebsiteLog && (
                                    <div className="d-flex justify-content-end">
                                        <div className="desktop-btn">
                                            <RdsButton
                                                type={"button"}
                                                size="small"
                                                colorVariant="primary"
                                                roundedButton={true}
                                                icon="refresh"
                                                class="me-2"
                                                iconHeight="15px"
                                                iconWidth="15px"
                                                iconFill={false}
                                                iconStroke={true}
                                                iconColorVariant="light"
                                                onClick={refreshData}
                                            ></RdsButton>
                                            <RdsButton
                                                type={"button"}
                                                label={t("Download All") || ""}
                                                onClick={downloadcsv}
                                                isOutline={true}
                                                colorVariant="primary"
                                                tooltipPlacement="top"
                                                size="small"
                                                icon="download"
                                                iconHeight="12px"
                                                iconWidth="12px"
                                                iconFill={false}
                                                iconStroke={true}
                                                iconColorVariant="primary"
                                            ></RdsButton>
                                        </div>
                                        <div
                                            className="mobile-btn position-fixed bottom-0 end-0 mb-3 me-5 custom-z-index">
                                            <RdsFabMenu
                                                listItems={listItems2}
                                                menuIcon="plus"
                                                colorVariant="primary"
                                                menuiconHeight="12px"
                                                menuiconWidth="12px"
                                            ></RdsFabMenu>
                                        </div>
                                    </div>
                                )}
                            </div>
                        </div>
                        <RdsNavtabs
                            navtabsItems={navtabsItems}
                            type={"tabs"}
                            fill={false}
                            justified={false}
                            activeNavTabId={activeNavTabId}
                            activeNavtabOrder={onchangetabs}
                        ></RdsNavtabs>
                        <div className="tab-content py-4" id="headerbar">
                            {activeNavTabId == "nav-cache" && (
                                <div
                                    className=" fade active show"
                                    id="nav-cache"
                                    role="tabpanel"
                                    aria-labelledby="nav-cache"
                                >
                                    <RdsCompCache
                                        cachedata={cache}
                                        recordsperpage={5}
                                        pagination={true}
                                        onclick={Delete}
                                        alignment={"end"}
                                    ></RdsCompCache>
                                </div>
                            )}
                            {activeNavTabId == "nav-websiteLogs" && (
                                <RdsCompWebsiteLog
                                    websiteLogData={websiteLogData}
                                    pagination={true}
                                    alignmentType="end"
                                    recordsPerPage={5}
                                    totalRecords={10}
                                />
                            )}
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Maintainance;
