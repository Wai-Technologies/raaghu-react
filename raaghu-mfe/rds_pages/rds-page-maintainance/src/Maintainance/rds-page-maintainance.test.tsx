import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { store } from "../../../../libs/state-management";
import Maintainance from "./Maintainance";
import { RdsButton, RdsFabMenu, RdsNavtabs } from "../../../rds-elements";
import { RdsCompAlertPopup, RdsCompCache, RdsCompWebsiteLog } from "../../../rds-components";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
      t: (key: string) => key,
    }),
    initReactI18next: { // add this line
      type: "3rdParty",
      init: () => {},
    },
  }));
jest.mock("react-router-dom", () => ({
    useNavigate: () => jest.fn(),
}));

global.location = {
    pathname: "../../../../libs/proxy/core",
  } as any;
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));
  jest.mock("../../../../../raaghu-components/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-components/src/index.ts"),
  }));
  jest.mock("../../../../../raaghu-elements/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-elements/src/index.ts"),
  }));
  jest.mock("../../../../libs/state-management/index.ts", () => ({
    store: {
      getState: jest.fn().mockReturnValue({
        persistedReducer: {
          configureStore: {
            configuration: {
              auth: {
                grantedPolicies: {},
              },
            },
          },
        },
      }),
      dispatch: jest.fn(),
      subscribe: jest.fn(),
    },
  }));
  
  // Mock the rds-elements and rds-components
  jest.mock("../../../../../raaghu-elements/src", () => ({
    RdsButton: jest.fn(),
    RdsOffcanvas: jest.fn(),
    RdsAlert: jest.fn(),
    RdsSearch: jest.fn(),
    RdsModal: jest.fn(),
    RdsLabel : jest.fn(),
    RdsInput : jest.fn(),
    RdsIcon : jest.fn(),
    RdsCheckbox : jest.fn(),
    RdsSelectList : jest.fn(),
    RdsBadge : jest.fn(),
    RdsFabMenu : jest.fn(),
    RdsNavtabs : jest.fn(),
  })); 
  
  jest.mock("../../../../../raaghu-components/src", () => ({
    RdsCompApiScopeBasicResource: jest.fn(),
    RdsCompDatatable: jest.fn(),
    RdsCompAlertPopup: jest.fn(),
    RdsOffcanvas:jest.fn(),
    RdsCompForgotPassword: jest.fn(),
    RdsCompFormsEmail: jest.fn(),
    RdsCompIconList: jest.fn(),
    RdsCompNewLanguage: jest.fn(),
    RdsCompWebsiteLog: jest.fn(),
    RdsCompCache: jest.fn(),
  }));
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useDispatch: jest.fn(),
  }));
  
  // Mock localStorage in your test setup
  const localStorageMock: Storage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
    length: 0, // Add length property
    key: jest.fn(), // Add key method
  };
  
  global.localStorage = localStorageMock;
  describe("Language Page", () => {
it("renders Language page correctly", () => {
    const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    render(
        <Provider store={store}>
            <Maintainance />
        </Provider>
    );
    document.querySelector('[label="Download All"]');
    document.querySelector('[button="Clear All"]');
});

it("Check maintainance page functions", () => {
  const DeleteAllCacheData = jest.fn();
  const listItems1 = {
    label: "Clear All",
    icon: "delete",
    onClick: DeleteAllCacheData,
  };
  const refreshData = jest.fn();
  const downloadcsv = jest.fn();
  const listItems2 = {
    label: "Download All",
    icon: "download",
    onClick: downloadcsv,
  };
  const navtabsItems = {
    label: "Refresh",
    icon: "refresh",
    onClick: refreshData,
  };
  const onchangetabs = jest.fn();
  const Delete = jest.fn();
  const cache = {
    id: "some-id",
    name: "some-name",
    type: "some-type",
    status: "some-status",
    lastUpdated: "some-lastUpdated",
    size: "some-size",
    action: "some-action",
  };
  const websiteLogData = {};
  const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
   const maintainance = render(
        <Provider store={store}>
            <Maintainance />
            <RdsButton label="Clear All" />
            <RdsCompAlertPopup alertID={"maintainance"}  onSuccess={DeleteAllCacheData}/>
            <RdsFabMenu listItems={[listItems1]} />
            <RdsButton  onClick={refreshData}/>
            <RdsButton label="Download All"  onClick={downloadcsv}/>
            <RdsFabMenu listItems={[listItems2]} />
            <RdsNavtabs navtabsItems={[{ ...navtabsItems, id: "some-id" }]} type={"tabs"}  activeNavtabOrder={onchangetabs}/>
            <RdsCompCache cachedata={[cache]} recordsperpage={0}  onclick={Delete}/>
            <RdsCompWebsiteLog websiteLogData={[websiteLogData]} pagination={false} alignmentType={undefined} totalRecords={0} recordsPerPage={0} />
        </Provider>
    );
    expect(maintainance).toMatchSnapshot();
});
afterEach(() => {
    jest.clearAllMocks(); // Clear mock calls between tests
    // Reset localStorage to a new object
    global.localStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
      length: 0,
      key: jest.fn(),
    };
  });
  })