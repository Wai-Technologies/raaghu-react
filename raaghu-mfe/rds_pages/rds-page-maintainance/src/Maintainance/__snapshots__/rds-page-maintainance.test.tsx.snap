// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Page Check maintainance page functions 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="card px-4 py-4 border-0 rounded-0 card-full-stretch"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12 text-end pb-3"
                >
                  <div
                    class="d-flex justify-content-end"
                  >
                    <div
                      class="desktop-btn"
                    />
                    <div
                      class="mobile-btn position-fixed bottom-0 end-0 mb-3 me-5 custom-z-index"
                    />
                  </div>
                </div>
              </div>
              <div
                class="tab-content py-4"
                id="headerbar"
              >
                <div
                  aria-labelledby="nav-cache"
                  class=" fade active show"
                  id="nav-cache"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="card px-4 py-4 border-0 rounded-0 card-full-stretch"
          >
            <div
              class="row"
            >
              <div
                class="col-md-12 text-end pb-3"
              >
                <div
                  class="d-flex justify-content-end"
                >
                  <div
                    class="desktop-btn"
                  />
                  <div
                    class="mobile-btn position-fixed bottom-0 end-0 mb-3 me-5 custom-z-index"
                  />
                </div>
              </div>
            </div>
            <div
              class="tab-content py-4"
              id="headerbar"
            >
              <div
                aria-labelledby="nav-cache"
                class=" fade active show"
                id="nav-cache"
                role="tabpanel"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
